make/src/rule.rs

```diff
@@ -64,100 +72,199 @@ impl Rule {
+                        continue;
+                    }
 
-            let mut command = Command::new(
-                env::var(DEFAULT_SHELL_VAR)
-                    .as_ref()
-                    .map(|s| s.as_str())
-                    .unwrap_or(DEFAULT_SHELL),
-            );
-            self.init_env(&mut command, macros);
-            command.args(["-c", recipe.as_ref()]);
-
-            let status = match command.status() {
-                Ok(status) => status,
-                Err(err) => {
-                    if ignore {
+                    // -t flag
+                    if touch {
                         continue;
+                    }
+                    // -q flag
+                    if quit {
+                        if up_to_date {
+                            process::exit(0);
+                        } else {
+                            process::exit(1);
+                        }
+                    }
+                }
+
+                // -s flag
+                if !silent {
+                    println!("{}", recipe);
+                }
+
+                let mut command = Command::new(
+                    env::var(DEFAULT_SHELL_VAR).as_ref().map(|s| s.as_str()).unwrap_or(DEFAULT_SHELL),
+                );
+
+                self.init_env(env_macros, &mut command, macros);
+                let recipe = self.substitute_internal_macros(target, recipe, &inout, self.prerequisites());
+                command.args(["-c", recipe.as_ref()]);
+
+                let status = match command.status() {
+                    Ok(status) => status,
+                    Err(err) => {
+                        if ignore {
+                            continue;
+                        } else {
+                            return Err(IoError(err.kind()));
+                        }
+                    }
+                };
+                if !status.success() && !ignore {
+                    // -S and -k flags
+                    if !terminate && keep_going {
+                        eprintln!(
+                            "make: {:?}",
+                            ExecutionError {
+                                exit_code: status.code(),
+                            }
+                        );
+                        break;
                     } else {
-                        return Err(IoError(err.kind()));
+                        return Err(ExecutionError {
+                            exit_code: status.code(),
+                        });
                     }
                 }
-            };
+            }
+
+            let silent = global_silent || rule_silent;
+            let touch = global_touch;
 
-            if !status.success() && !ignore {
-                return Err(ExecutionError {
-                    exit_code: status.code(),
-                });
+            // -t flag
+            if touch {
+                if !silent {
+                    println!("touch {target}");
```

Comment:
gettext

Reply:
I can't understand how should I use `gettext` here

Reply:
I can't understand how should I use `gettext` here

Reply:
```suggestion
                    println!("{} {target}", gettext("touch"));
```
something like this